### PR TITLE
Add the options apply_live to rpm_ostree_pkg

### PR DESCRIPTION
**Pull request by shios86**
##### SUMMARY

Feature request for #8950 #3670 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
rpm_ostree_pkg

### CODE REVIEW

File: changelogs/fragments/9167-rpm_ostree_pkg-apply_live.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+- rpm_ostree_pkg - added the options apply_live and reboot
```
Comment (russoz) [inline changelogs/fragments/9167-rpm_ostree_pkg-apply_live.yml:2]:
```suggestion
- rpm_ostree_pkg - added the options apply_live and reboot (https://github.com/ansible-collections/community.general/pull/9167).
```

File: plugins/modules/rpm_ostree_pkg.py [hunk 2]
```diff
@@ -40,6 +40,20 @@
       choices: [ 'absent', 'present' ]
       default: 'present'
       type: str
+    apply_live:
+      description:
+      - Adds the options C(--apply-live) when O(state) is V(present)
```
Comment (russoz) [inline plugins/modules/rpm_ostree_pkg.py:45]:
```suggestion
      - Adds the options C(--apply-live) when O(state) is V(present).
```

Comment (russoz) [reply plugins/modules/rpm_ostree_pkg.py:45]: Or maybe:
```suggestion
      - Adds the options C(--apply-live) when O(state=present).
```

File: plugins/modules/rpm_ostree_pkg.py [hunk 3]
```diff
@@ -40,6 +40,20 @@
...
+      - Option is ignored when O(state) is V(absent)
+      - For more information, please see U(https://coreos.github.io/rpm-ostree/apply-live/)
```
Comment (russoz) [inline plugins/modules/rpm_ostree_pkg.py:47]:
```suggestion
      - Option is ignored when O(state) is V(absent).
      - For more information, please see U(https://coreos.github.io/rpm-ostree/apply-live/).
```

File: plugins/modules/rpm_ostree_pkg.py [hunk 4]
```diff
@@ -40,6 +40,20 @@
...
+      type: bool
+      default: false
+      required: false
+    reboot:
+      description:
+      - Reboot the machine after making changes to the overlay
```
Comment (russoz) [inline plugins/modules/rpm_ostree_pkg.py:53]:
```suggestion
      - Reboot the machine after making changes to the overlay.
```

File: plugins/modules/rpm_ostree_pkg.py [hunk 5]
```diff
@@ -137,6 +163,12 @@ def ensure(self):
             results['action'] = 'uninstall'
             cmd.append('uninstall')
 
+        # Add the options to the command line
+        if self.params['apply_live'] and self.state in ('present'):
```
Comment (russoz) [inline plugins/modules/rpm_ostree_pkg.py:167]: Why not:
```suggestion
        if self.params['apply_live'] and self.state == 'present':
```
?

**Review by russoz — COMMENTED** [review]
Hi @shios86 , thanks for your contribution!

There are some adjustments required, but the overall feature looks good to me.

Cheers

File: plugins/modules/rpm_ostree_pkg.py (hunk 5 above)
Comment (shios86) [reply plugins/modules/rpm_ostree_pkg.py:167]: I took it from line 159 to stay uniform with the rest of the file but yes, it could be simplified

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution!

File: plugins/modules/rpm_ostree_pkg.py (hunk 5 above)
Comment (felixfontein) [reply plugins/modules/rpm_ostree_pkg.py:167]: Instead of repeating mistakes from existing code, please fix lines 159 and 162 as well.

**Comment by shios86** [comment]
I might have pushed the option `reboot` a little bit too fast.
Making rpm-ostree reboot the machine will fail the job will close ssh connection before ansible does.
I have removed the `reboot` option from the file, keeping only `apply_live`.

**Comment by russoz** [comment]
@shios86 if possible, you could add a return value, something like `needs_reboot`, so that the user of the module might use it to trigger a reboot later in their playbook.

**Comment by shios86** [comment]
@russoz I've added your suggestion into the commit. I've only tested this on Fedora CoreOS41, although `rpm-ostree` could have different output on other distribution.

**Review by russoz — APPROVED** [review]
Thanks for that, LGTM

File: plugins/modules/rpm_ostree_pkg.py [hunk 6]
```diff
@@ -104,6 +118,12 @@
     returned: always
     type: str
     sample: 'rpm-ostree uninstall --allow-inactive --idempotent --unchanged-exit-77 nfs-utils'
+needs_reboot:
+    description: Determine if machine needs a reboot to apply current changes.
+    returned: always
```
Comment (felixfontein) [inline plugins/modules/rpm_ostree_pkg.py:123]:
```suggestion
    returned: success
```
This won't get added to the output if the module fails :)

Comment (shios86) [reply plugins/modules/rpm_ostree_pkg.py:123]: My mistake, fixing now

**Comment by felixfontein** [comment]
@shios86 thanks for your contribution!
@russoz thanks for reviewing!

**Pull request merged**